code/modules/antagonists/antagonist_helpers.dm

```diff
@@ -1,25 +1,33 @@
+/// An associative list of all antagonist IDs, associated with a list of all antagonist datums of that ID.
+var/list/antagonists = list()
+
 /**
- * Gets a list of all antagonists of ID role_id.
- * Returns a list if any datums are present, or null if none are.
+ * Gets a list of all antagonist datums of ID role_id, or of all IDs if no ID is specified.
+ * Returns a list of all antagonist datums. If no antagonist datums could be found, returns an empty list.
  */
-/proc/get_all_antagonists(role_id, include_pseudo = FALSE)
+/proc/get_all_antagonists(antagonist_role_id)
 	. = list()
-	for (var/datum/mind/mind in ticker.minds)
-		for (var/datum/antagonist/antag in mind.antagonists)
-			if (antag.id == role_id && (!antag.pseudo || include_pseudo))
-				. += antag
-				break
+
+	if (antagonist_role_id)
+		return antagonists["[antagonist_role_id]"]
+
+	for (var/antagonist_type in concrete_typesof(/datum/antagonist))
+		var/datum/antagonist/A = antagonist_type
+		var/list/datum/antagonist/antagonist_roles = antagonists["[initial(A.id)]"]
+		if (length(antagonist_roles))
+			. += antagonist_roles
```

Comment:
Am I missing something or could we just iterate over `antagonists` and join the values into one list?

Reply:
Yes, that would be a far better method than the above.
Thank you!